aten/src/ATen/core/ivalue.h

```diff
@@ -707,6 +707,13 @@ struct C10_EXPORT ivalue::Object final : c10::intrusive_ptr_target {
   const std::vector<IValue>& slots() const {
     return slots_;
   }
+  std::shared_ptr<ClassType> type() const {
+    return type_;
+  }
+  void appendSlot(IValue v) {
```

Comment:
I don't like that this relies on the callers to maintain coherency between the class type's attributes and an instance object's slots. Especially since type->object is a 1:many mapping.

Can we instead augment "setSlot()" to resize the vector if necessary (perhaps with an assert that validates that the slot # makes sense for the class type), and force callers to use the class type's `getAttributeSlot` to find the correct slot?

Reply:
Much better, I will do that.